prolif/interactions/interactions.py

```diff
@@ -131,6 +133,140 @@ def __init__(
 )
 
 
+class ImplicitHBAcceptor(Distance):
+    """Implicit hydrogen bond acceptor interaction
+
+    Parameters
+    ----------
+    acceptor : str
+        SMARTS for ``[Acceptor]``
+    donor : str
+        SMARTS for ``[Donor]-[Hydrogen]``
+    tolerance_dev_aaa : float
+        Tolerance for the deviation from the ideal acceptor atom's angle
+    tolerance_dev_daa : float
+        Tolerance for the deviation from the ideal donor atom's angle
+    tolerance_dev_apa : float
+        Tolerance for the deviation from the ideal acceptor plane angle
+    tolerance_dev_dpa : float
+        Tolerance for the deviation from the ideal donor plane angle
+    distance : float
+        Distance threshold between the acceptor and donor atoms
+
+    """
+
+    def __init__(
+        self,
+        acceptor: str = (
+            "[#7&!$([nX3])&!$([NX3]-*=[O,N,P,S])&!$([NX3]-[a])&!$([Nv4&+1]),"
+            "O&!$([OX2](C)C=O)&!$(O(~a)~a)&!$(O=N-*)&!$([O-]-N=O),o+0,"
+            "F&$(F-[#6])&!$(F-[#6][F,Cl,Br,I])]"
+        ),
+        donor: str = "[$([O,S;+0&h]),$([N;v3,v4&+1&h]),n+0&h]",
+        distance: float = 3.5,
+        tolerance_dev_aaa: float = 45,
+        tolerance_dev_daa: float = 45,
+        tolerance_dev_apa: float = 90,
+        tolerance_dev_dpa: float = 45,
+    ) -> None:
+        super().__init__(lig_pattern=acceptor, prot_pattern=donor, distance=distance)
+        self.tolerance_dev_aaa = tolerance_dev_aaa
+        self.tolerance_dev_daa = tolerance_dev_daa
+        self.tolerance_dev_apa = tolerance_dev_apa
+        self.tolerance_dev_dpa = tolerance_dev_dpa
+
+    def detect(self, lig_res, prot_res):
+        """Detect implicit hydrogen bond acceptor interactions.
+
+        Parameters
+        ----------
+        lig_res : Residue
+            Ligand residue.
+        prot_res : Residue
+            Protein residue.
+
+        Yields
+        ------
+        InteractionMetadata
+            Metadata for the detected interaction.
+
+        """
+        for interaction_data in super().detect(lig_res, prot_res):
+            # Check if the interaction geometry is valid
+            if self.check_geometry(
+                interaction_data,
+                lig_res,
+                prot_res,
+                max_dev_aaa=self.tolerance_dev_aaa,
```

Comment:
all those tolerance parameters are available to any method in the class, there's no need to pass them around like that.
Usually if you start passing the attributes of an object to a method of that same object there's something wrong with the design